docs/microservices-framework/event-sources/on-validation-error.md

```diff
@@ -0,0 +1,121 @@
+# Validation Error
+
+- A validation error occurs when data fails to meet the defined criteria or constraints during the validation process.
+- Validation is the process of ensuring that data conforms to a set of rules or conditions before it is accepted or processed by a system. 
+
+## Request Validation
+- Verifying that incoming API requests have the required parameters and that those parameters meet specific criteria like data types.  If the specified criteria are not met, it results in a request validation error. 
+- For more info about Request Validation and its applications, refer [this](/docs/microservices-framework/event-sources/schema-validation.md#request-schema-validation)
+
+:::tip Note
+The `on_request_validation_error` and `on_response_validation_error` handlers are used to override the default errors thrown by the framework (specifically, schema validation errors) and allow developers to customize errors based on their requirements.
+:::
+
+### on_request_validation_error
+
+- To customize the error response in cases where request schema validation fails, you can utilize the `on_request_validation_error` handler. Demonstrating its use in the below example
+
+### Example
+
+```yaml
+"http.get./helloworld":
+  fn: helloworld
+  on_request_validation_error: request_error #can be fn path, or a series of tasks
+  params:
+  - name: num_1
+    in: query
+    required: false
+    schema:
+      type: number
+  - name: num_2
+    in: query
+    required: true
+    schema:
+      type: number
+
+  responses:
+    200:
+      content:
+        application/json:
+          schema:
+            type: object
+```
+functions/request_error.yaml
+```yaml 
+summary: Customizing request_validation_error
+tasks:
+  - id: request_validation_error
+    fn: com.gs.transform
+    args: {success: false, code: 400, message: "request error", data: {req: request schema validation failed}}
```

Comment:
Please run this code and show examples of running code.. you will get the data of request error failure in the ctx.inputs.data object. inspect that object. return that object in your response transformation. developer needs to know where exactly can he find what error happened, when writing the validation error transformation